http/server_sent_event.ts

```diff
@@ -0,0 +1,411 @@
+
+  /** Close the target, refusing to accept any more events. */
+  close(): Promise<void>;
+
+  /** Send a comment to the remote connection.  Comments are not exposed to the
+   * client `EventSource` but are used for diagnostics and helping ensure a
+   * connection is kept alive.
+   *
+   * ```ts
+   * import { SSEStreamTarget } from "https://deno.land/std@$STD_VERSION/http/server_sent_event.ts";
+   * import { serve } from "https://deno.land/std@$STD_VERSION/http/server.ts";
+   *
+   * await serve((request) => {
+   *   const target = new SSEStreamTarget();
+   *   target.dispatchComment("this is a comment");
+   *   return target.asResponse();
+   * }, { port: 8000 });
+   * ```
+   */
+  dispatchComment(comment: string): boolean;
+
+  /** Dispatch a message to the client.  This message will contain `data: ` only
+   * and be available on the client `EventSource` on the `onmessage` or an event
+   * listener of type `"message"`. */
+  dispatchMessage(data: unknown): boolean;
+
+  /** Dispatch a server sent event to the client.  The event `type` will be
+   * sent as `event: ` to the client which will be raised as a `MessageEvent`
+   * on the `EventSource` in the client.
+   *
+   * Any local event handlers will be dispatched to first, and if the event
+   * is cancelled, it will not be sent to the client.
+   *
+   * ```ts
+   * import {
+   *   ServerSentEvent,
+   *   SSEStreamTarget,
+   * } from "https://deno.land/std@$STD_VERSION/http/server_sent_event.ts";
+   * import { serve } from "https://deno.land/std@$STD_VERSION/http/server.ts";
+   *
+   * await serve((request) => {
+   *   const target = new SSEStreamTarget();
+   *   const evt = new ServerSentEvent("ping", "hello");
+   *   target.dispatchEvent(evt);
+   *   return target.asResponse();
+   * }, { port: 8000 });
+   * ```
+   */
+  dispatchEvent(event: ServerSentEvent): boolean;
+
+  /** Dispatch a server sent event to the client.  The event `type` will be
+   * sent as `event: ` to the client which will be raised as a `MessageEvent`
+   * on the `EventSource` in the client.
+   *
+   * Any local event handlers will be dispatched to first, and if the event
+   * is cancelled, it will not be sent to the client.
+   *
+   * ```ts
+   * import {
+   *   ServerSentEvent,
+   *   SSEStreamTarget,
+   * } from "https://deno.land/std@$STD_VERSION/http/server_sent_event.ts";
+   * import { serve } from "https://deno.land/std@$STD_VERSION/http/server.ts";
+   *
+   * await serve((request) => {
+   *   const target = new SSEStreamTarget();
+   *   const evt = new ServerSentEvent("ping", "hello");
+   *   target.dispatchEvent(evt);
+   *   return target.asResponse();
+   * }, { port: 8000 });
+   * ```
+   */
+  dispatchEvent(event: CloseEvent | ErrorEvent): boolean;
+}
+
+/** An implementation of {@linkcode ServerSentEventTarget} that provides a
+ * readable stream as a body of a response to establish a connection to a
+ * client. */
+export class SSEStreamTarget extends EventTarget
```

Comment:
Maybe we should rename this to `ServerSentEventStreamTarget` for consistency? (SSE acronym doesn't look used in other places)

(Also if we use `SSE` acronym, it should be `Sse`. We decided not to use ALL CAPS acronyms in public API https://github.com/denoland/deno_std/pull/2582)

Reply:
Agreed, I have renamed it.